Documentation/CraftingNuGetPackages.md

```diff
@@ -0,0 +1,69 @@
+
+- Unity Version (ie Major.Minor)
+- Platform being targeted (UWP, Android, Standalone, etc).
+- .NET Target Framework (4.6, .NET Standard 2.0)
+- Compilation define symbols specified
+- For UWP Platform we know Min and Target UWP SDK
+
+Because this stage is not aware of the target device, and happens before we build the player output (to Android Studio, XCode or Visual Studio) we don't have the following information:
+
+- The target platform of the device (ARM, x86, x64)
+
+This means that we don't know the target platform of the device and thereby can't bring in a specific native DLL (ARM, x86, x64). Thus for NuGet packages with Native code, we must add additional logic to bring in every DLL as well as Unity specific .meta files.
+
+### Meta Files
+
+MSBuildForUnity performs meta file generation for .NET libraries, and there are plans to support processing Native DLLs as well, however this is not possible yet. For this reasons, packages with Native DLLs that must be brought into Unity must be accompanied with .meta files configured for those packages.
+
+### Properties Provided Through MSBuildForUnity
+
+The following properties are either passed through as the build CLI argument, or available in the Unity Project root's `MSBuildForUnity.Common.props` file.
+
+- **UnityConfiguration:** Whether we are building for `InEditor` or `Player`. (**LIMITATION** we only support `InEditor` for now; [Issue #59](https://github.com/microsoft/MSBuildForUnity/issues/59))
+- **UnityPlatform:** The platform that we are building for, the values match Unity's `BuildTarget` enum ([documentation](https://docs.unity3d.com/ScriptReference/BuildTarget.html))
+- **MSBuildForUnityVersion:** The `{Major}.{Minor}.{Patch}` version of MSBuildForUnity, it can be used in comparison.
+- **MSBuildForUnityDefaultOutputPath:** The default output path for the Unity project's dependencies.
+  > **NOTE:** Don't rely on this unless you absolutely need to, rely on default MSBuild "copy-to-output" behavior.
+- **TargetFramework:** By default, MSBuild will make TargetFramework available.
+  > MSBuild will also automatically bring in the DLLs from the NuGet packages' matching target framework folder.
+- **UnityMajorVersion:** The major version of Unity (ie. 2018, 2019, etc)
+- **UnityMinorVersion:** The minor version of Unity (ie. 1, 2, 3, 4)
+
+## Supporting MSBuildForUnity in your NuGet Package
+
+As mentioned in the [Quickstart Summary](##Quickstart%20Summary), to augment your NuGet package, you should do the following:
+
+1. Add [Build Targets or Props](https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#include-msbuild-props-and-targets-in-a-package) file to your package.
```

Comment:
nit: it would be great if each one of these steps could call out a quick blurb of why users are doing this compared to just the step to take. That may better convey the intentions of the actions they are taking to help build a better understanding of the process. Just having a step by step with no explanation of why can be confusing.

One example:
2. Add necessary .meta files for your Native libraries.

This could be

2. Add necessary .meta files for your Native libraries so that MSBuildForUnity and Unity can resolve their intended platform (x86, x64, arm, etc).

Reply:
+1 for this

I think that an example of what you have to do would also greatly help - i.e. this could be pointing to an existing example of someone who has done this, or having a snippet of code/XML to make things clearer.

Basically you did this for step 5 which was awesome - it would be helpful to have these for the other steps as well.